base/compiler/ssair/ir.jl

```diff
@@ -428,22 +428,26 @@ struct IRCode
     cfg::CFG
     new_nodes::NewNodeStream
     meta::Vector{Expr}
+    assume_bindings_static::Bool # XXX propagate `interp::AbstractInterpreter` here?
```

Comment:
I don't think adding this field here is reasonable. We should find some other way to propagate this information.

Reply:
Yeah.. Would it be too hacky to temporarily keep `Expr(:assume_bindings_static, true)` in `meta::Vector{Expr}`?

Reply:
I think that would be fine.

Reply:
I'm not very happy with that solution.

Assuming that some implicit subset of InferenceParams have been copied into the `meta` vector is very subtle and bakes implicit pipeline ordering into the construction of an IRCode.

For example, it looks like a caller of `ssa_inlining_pass!` will have their Interpreter's parameters ignored when `stmt_effect_flags` are computed internally (despite the fact that `InliningState` contains an `interp` which has params).

Is the caller supposed to provide a "meta-expanded" IRCode? How are they supposed to know which settings will be ignored?